### DISCUSSION TITLE
Failed to parse source map from...

### DISCUSSION THREAD

**Discussion by JoeySpen** [jeffreylanters/react-unity-webgl#283] [Help Wanted!] [answered]
Hi, I'm trying to get started with the library but I'm having trouble. I'm running npm install react-unity-webgl@7.x in my create-react-app project, but I'm getting an error when trying to import from the Unity library.

`WARNING in ./node_modules/react-unity-webgl/distribution/utils/generate-unity-instance-parameters.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/joeyspen/Documents/swords/node_modules/react-unity-webgl/source/utils/generate-unity-instance-parameters.ts' file: Error: ENOENT: no such file or directory, open '/home/joeyspen/Documents/swords/node_modules/react-unity-webgl/source/utils/generate-unity-instance-parameters.ts'
 @ ./node_modules/react-unity-webgl/distribution/components/unity.js 16:43-97
 @ ./node_modules/react-unity-webgl/distribution/index.js 8:14-43
 @ ./src/App.js 6:0-56 8:25-37
 @ ./src/index.js 7:0-24 11:33-36`

I've tried this on both a linux and windows machine. There is no /source directory within my react-unity-webgl folder

**Comment by BattleBlockz** [comment]
I am having this same issue. I have tried the sample files as well as my own build and get this exact same problem. All I am shown when I see the react build is a blank white page. I tried to rollback my unity from 2020 as well as the module and no difference. One thing I personally notice is the files created by unity after build on minimal and default are data.unityweb, js.unityweb, and wasm.unityweb instead of whats stated in examples.

Not sure what the issue is here but even with some tweaking I could not find a fix!

**Comment by jeffreylanters** [comment]
Which version of the module are you using, and what version and language of create react app?

**Reply by BattleBlockz** [reply]
Using the 8.x version of the module with npm 8.3. Javascript is the language I am using. The error I am getting is

`WARNING in ./node_modules/react-unity-webgl/distribution/library/unity-context.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from "FILEPATH" file: Error: ENOENT: no such file or directory. `
 `@ ./node_modules/react-unity-webgl/distribution/library/unity-context.js 174:21-46`
 `@ ./node_modules/react-unity-webgl/distribution/index.js 10:22-56`
` @ ./src/App.js 8:0-56 10:25-37 33:35-40`
 `@ ./src/index.js 7:0-24 11:33-36`

In the console it states 

`"Uncaught SyntaxError: Unexpected token '<' 586cb24……cb17d.js.unityweb:1"`

I attempted to add the loader/framework scripts to the index in hopes maybe it would still load but no luck.

**Reply by jeffreylanters** [reply]
This seems to be a duplicate of #57

**Comment by BattleBlockz** [comment]
I believe I have figured out the problem and I can't believe how slow I am for this mistake. My build folder was in my src and not in public. It was as simple as that. I'm guessing the module was looking to grab the files from public folder but couldn't find them which triggered the warning. Sometimes I question my knowledge in programming for mistakes like this haha.

So @JoeySpen I recommend making sure your build folder is in public where your index.html and manifest.json is located.

**Reply by jeffreylanters** [reply]
No worries, we're all learning new things every day. I'm glad you solved it, best of luck with your project!

**Comment by johnbatista1223** [comment]
i  have my build directory in public and still getting the same issue

**Reply by Caydo** [reply]
Likewise here.

- Installed NodeJS 16.13.1
- Ran npx create-react-app my-app
- Ran npm install react-unity-webgl@8.x
- Replaced the contents of App.js with the example snippet under Getting Started

Hitting the initial error regardless of putting the build directory being in the public folder.

Double-checked that the URLs matched what was in the build directory.

So it's "my-app/public/build/build.loader.js" , etc. with the full contents of App.js looking like this:

```
import React from "react";
import Unity, { UnityContext } from "react-unity-webgl";

const unityContext = new UnityContext({
  loaderUrl: "build/build.loader.js",
  dataUrl: "build/build.data",
  frameworkUrl: "/build/build.framework.js",
  codeUrl: "build/build.wasm",
});

function App() {
  return <Unity unityContext={unityContext} />;
}

export default App;
```

**Reply by jeffreylanters** [reply]
Thanks, I successfully reproduced the error. I'm investigating it right now.

**Reply by Caydo** [reply]
Oh, wow, appreciate the swift reply! Have used a previous iteration of the project and it's awesome!

**Comment by jeffreylanters** [comment, marked as answer]
> @Caydo @JoeySpen @johnbatista1223 

I've released version v8.7.2 which should resolve this issue. Could you try this and let this board know if it worked for you? Thanks!

**Reply by Caydo** [reply]
Looks like that fixed the errors, thanks! Not getting the build to display in the browser, but I'll see if there's something I missed in setup or a copy-paste error or something.

**Reply by jeffreylanters** [reply]
Awesome! Check out the [Test repository](https://github.com/jeffreylanters/react-unity-webgl-tests) for a complete example project, or open a new discussion if you need any further help. Just let me know! :)

**Reply by 11samype** [reply]
@jeffreylanters Sorry to bump such an old topic. Would it be possible to have this fixed on 7.x versions? My project was built in Unity 2019.x and has not been updated and was hoping to avoid moving to something newer.
